server/services/v1/auth_provider.go

```diff
@@ -327,6 +414,33 @@ func getAccessTokenUsingClientCredentials(req *api.GetAccessTokenRequest, a *Aut
 		if err != nil {
 			return nil, api.Errorf(api.Code_INTERNAL, "Failed to parse external response: reason = %s", err.Error())
 		}
+		// cache it
+		tx, err := a.txMgr.StartTx(ctx)
+		if err != nil {
+			log.Warn().Err(err).Msg("Failed to start tx")
+			return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = could not process cache")
```

Comment:
We could define a static error for this, so as it's used multiple time below

Reply:
created helper and deduplicated code.